File: examples/python/api_client.py
#!/usr/bin/env python3
"""HTTP API: issue (7-day), get, verify, revoke.

Env: BASE_URL (default http://localhost:8080)
"""
from __future__ import annotations

import os
from datetime import datetime, timedelta, timezone

import requests

BASE = os.environ.get("BASE_URL", "http://localhost:8080").rstrip("/")


def main() -> None:
    exp = (datetime.now(timezone.utc) + timedelta(days=7)).isoformat()
    r = requests.post(
        f"{BASE}/v1/licenses",
        json={"expires_at": exp},
        timeout=15,
    )
    r.raise_for_status()
    data = r.json()
    lic_id = data["id"]
    lic_key = data["license_key"]
    print("issued id:", lic_id)

    r2 = requests.get(f"{BASE}/v1/licenses/{lic_id}", timeout=15)
    print("get:", r2.status_code)

    r3 = requests.post(
        f"{BASE}/v1/licenses/verify",
        json={"license_key": lic_key},
        timeout=15,
    )
    print("verify:", r3.json())

    r4 = requests.post(f"{BASE}/v1/licenses/{lic_id}/revoke", timeout=15)
    print("revoke:", r4.status_code, r4.text)


if __name__ == "__main__":
    main()

File: examples/javascript/api_client.mjs
/**
 * HTTP API: issue (7-day), get, verify, revoke.
 *
 * Env: BASE_URL (default http://localhost:8080)
 * Node 18+
 */

const BASE = (process.env.BASE_URL ?? "http://localhost:8080").replace(/\/$/, "");

const exp = new Date(Date.now() + 7 * 24 * 60 * 60 * 1000).toISOString();
const issueRes = await fetch(`${BASE}/v1/licenses`, {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({ expires_at: exp }),
});
if (!issueRes.ok) throw new Error(`issue ${issueRes.status} ${await issueRes.text()}`);
const { id, license_key: licenseKey } = await issueRes.json();
console.log("issued id:", id);

const getRes = await fetch(`${BASE}/v1/licenses/${id}`);
console.log("get:", getRes.status);

const verifyRes = await fetch(`${BASE}/v1/licenses/verify`, {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({ license_key: licenseKey }),
});
console.log("verify:", await verifyRes.json());

const revokeRes = await fetch(`${BASE}/v1/licenses/${id}/revoke`, { method: "POST" });
console.log("revoke:", revokeRes.status, await revokeRes.text());
